src/cache/sqlite.ts

```diff
@@ -74,7 +77,17 @@ class SQLiteAdapter {
 
               return responses;
             });
-        }, Promise.resolve());
+        }, Promise.resolve())
+          .then((response) => {
+            return db.close()
```

Comment:
My main concern is with the potential performance hits of closing and reopening the `db` each time. For correctness, this seems fine but let's run a performance test soon to figure out how well the db operations scale.